### PR TITLE
add more tslint checks

### PR DESCRIPTION
**Pull request by ericvw**
- enforece PascalCased class names
- enforece properly checks when using `for ... in` constructs
- interfaces names must begin with capital `I`
- no unused expressions (i.e., expressions that contain side-effects)
- enforce type definitions in function parameters, properties for
  interface declarations, and member variable declarations

Affected files by the added checks have been updated appropriately.

### CODE REVIEW

File: lib/middleware/request-handler.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export function elevateRequest (req: Interface.IOurRequest,
         });
     };
 
-    res.sendError = function(err, where, reason): Promise<void> {
+    res.sendError = function(err: any, where: string, reason: any): Promise<void> {
```
Comment (ericlu88) [inline lib/middleware/request-handler.ts:90]:
``` typescript
err: Error, where: string, reason: string
```

File: lib/config.ts [hunk 2]
```diff
@@ -191,7 +191,7 @@ export function get(name: string): any
     // Build options object
     // Bunyan logger options
     // Override default bunyan response serializer
-    bunyan.stdSerializers['res'] = function(res): any {
+    bunyan.stdSerializers['res'] = function(res: any): any {
```
Comment (ericlu88) [inline lib/config.ts:194]: `res: interface.IOurResponse` ?

File: lib/middleware/request-handler.ts (hunk 1 above)
Comment (ericlu88) [inline lib/middleware/request-handler.ts:90]: I will vote for getting rid of `res.sendError` and let `restify` handle the sending of errors.(The method has 0 usage in our codebase now).

Comment (ericvw) [inline lib/middleware/request-handler.ts:90]: I concur.

File: lib/config.ts (hunk 2 above)
Comment (ericvw) [inline lib/config.ts:194]: Decided to keep with `any` because it can by `any`thing.

File: lib/middleware/request-handler.ts (hunk 1 above)
Comment (ericvw) [inline lib/middleware/request-handler.ts:90]: Let's leave it for now for the purposes of this PR and then in a follow-up remove this.

**Comment by ericvw** [comment]
Based on discussion, this is ready for merge if approved by reviewers.

**Comment by ericlu88** [comment]
LGTM.

**Pull request merged**